grammar.js

```diff
@@ -905,6 +925,99 @@ module.exports = grammar({
       ')',
     ),
 
+    window_partition_by_expression: $ => seq(
+        $.keyword_partition_by,
+        $.identifier,
+    ),  
+
+    window_order_by_expression: $ => seq(
+        $.keyword_order_by,
+        $.identifier,
+        optional(
+            seq(
+                choice(
+                    $.direction,
+                    $.keyword_using,
+                ),
+                optional(
+                    seq(
+                        $.keyword_nulls,
+                        choice(
+                            $.keyword_first,
+                            $.keyword_last,
+                        ),
+                    ),
+                ),
+            ),
+        ),
+    ),  
+
+    _frame_definition: $ => seq(
+        choice(
+            $.keyword_unbounded_preceding,
+            seq(
+                alias($._number, $.literal),
+                $.keyword_preceding,
+            ),
+            $.keyword_current_row,
+            seq(
+                alias($._number, $.literal),
+                $.keyword_following,
+            ),
+            $.keyword_unbounded_following,
+        ),
+    ),
+
+    window_frame_clause: $ => seq(
+        choice(
+            $.keyword_range,
+            $.keyword_rows,
+            $.keyword_groups,
+        ),
+        optional(
+            $.keyword_between,
+        ),
+        $._frame_definition,
+        $.keyword_and,
+        $._frame_definition,
+        optional(
+            choice(
+                $.keyword_exclude_current_row,
+                $.keyword_exclude_group,
+                $.keyword_exclude_ties,
+                $.keyword_exclude_no_others,
+            ),
+        ), 
+    ),
+
+    window_clause: $ => seq(
+        $.keyword_window,
+        $.identifier,
+        $.keyword_as,
+        $.window_specification,
+    ),
+
+    window_specification: $ => seq(
+        '(',
+        repeat(
+            choice(
```

Comment:
at least with Postgres  you get up to one of each of `partition by`, `order by`, and frame clause in that order. MySQL docs appear to confirm ordering, and I don't think we need to worry about multidimensional windowing, so this can be a seq() of optional()s.